pandas/src/datetime/np_datetime_strings.c

```diff
@@ -546,12 +558,19 @@ parse_iso_8601_datetime(char *str, int len,
         bestunit = PANDAS_FR_Y;
         goto finish;
     }
-    else if (*substr == '-') {
-        ++substr;
-        --sublen;
-    }
-    else {
-        goto parse_error;
+    else if (!isdigit(*substr)) {
+        for (i = 0; i < valid_sep_len; ++i) {
+            if (*substr == valid_sep[i]) {
+                has_sep = 1;
+                sep = valid_sep[i];
+                ++substr;
+                --sublen;
+                break;
+            }
+        }
+        if (i == valid_sep_len) {
+            goto parse_error;
```

Comment:
This first check (sep between year and month) validates its in the list - the second check (between month/day) is required to match `sep`